### PR TITLE
[multistage] clean up aggregate v1 function for v2 planner

### PR DESCRIPTION
**Pull request by walterddr**
Summary
===
clean up hint-based aggregate node representation, see #11034 

Detail
===
- consolidate mode / agg / hint all into just one AggType enum
- remove all unnecessary castings and boolean extractions.

### CODE REVIEW

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java [hunk 1]
```diff
@@ -29,12 +29,29 @@ public class PinotHintOptions {
...
 
+  public static class InternalAggregateOptions {
+    public static final String AGG_TYPE = "agg_type";
+    public enum AggType {
+      DIRECT,
+      LEAF,
+      INTERMEDIATE,
+      FINAL,
+      REDUCE // NOT SUPPORTED
+    }
```
Comment (walterddr) [inline pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java:49]: these are the only necessary MODE we needed

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java [hunk 2]
```diff
@@ -29,12 +29,28 @@ public class PinotHintOptions {
...
+  public static final String INTERNAL_AGG_OPTIONS = "aggOptionsInternal";
+
   private PinotHintOptions() {
     // do not instantiate.
   }
 
+  public static class InternalAggregateOptions {
+    public static final String AGG_TYPE = "agg_type";
+    public enum AggType {
+      DIRECT,
```
Comment (xiangfu0) [inline pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java:44]: Can you add more comments on the type to distinguish `DIRECT` and `LEAF`?

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java [hunk 3]
```diff
@@ -248,12 +227,12 @@ private RelNode makeNewIntermediateAgg(RelOptRuleCall ruleCall, Aggregate oldAgg
...
   private static void convertIntermediateAggCall(RexBuilder rexBuilder, Aggregate oldAggRel, int oldCallIndex,
       AggregateCall oldCall, List<AggregateCall> newCalls, Map<AggregateCall, RexNode> aggCallMapping,
-      boolean isLeafStageAggregationPresent, List<Integer> argList, PinotLogicalExchange exchange) {
+      AggType aggType, List<Integer> argList, PinotLogicalExchange exchange) {
     final int nGroups = oldAggRel.getGroupCount();
     final SqlAggFunction oldAggregation = oldCall.getAggregation();
 
     List<Integer> newArgList;
-    if (isLeafStageAggregationPresent) {
+    if (AggType.INTERMEDIATE.equals(aggType) || AggType.FINAL.equals(aggType)) {
```
Comment (xiangfu0) [inline pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java:235]: worth a static method `isLeafStageAggregation()`  for this?

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java [hunk 4]
```diff
@@ -29,12 +29,28 @@ public class PinotHintOptions {
...
+   */
+  public static final String INTERNAL_AGG_OPTIONS = "aggOptionsInternal";
+
   private PinotHintOptions() {
     // do not instantiate.
   }
 
+  public static class InternalAggregateOptions {
+    public static final String AGG_TYPE = "agg_type";
+    public enum AggType {
```
Comment (xiangfu0) [inline pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java:43]: suggest add boolean methods like: isFinal(), isLeaf()... for comparison simplicity.

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/AggregateNode.java [hunk 5]
```diff
@@ -115,29 +72,13 @@ public List<RelHint> getRelHints() {
...
-  public boolean isLeafStage() {
-    return _aggregationStage == AggregationStage.LEAF;
+  public PinotHintOptions.InternalAggregateOptions.AggType getAggType() {
+    return _aggType;
   }
 
   @Override
   public String explain() {
-    return "AGGREGATE";
+    return "AGGREGATE" + _aggType;
```
Comment (xiangfu0) [inline pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/AggregateNode.java:81]: `"AGGREGATE_" + _aggType` ?

**Review by xiangfu0 — APPROVED** [review]
Very neat!

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java (hunk 3 above)
Comment (walterddr) [reply pinot-query-planner/src/main/java/org/apache/calcite/rel/rules/PinotAggregateExchangeNodeInsertRule.java:235]: yeah i tried. this is not the same as "isLeafStage" b/c the treatment during process input block and produce output block as different. so it has to be directly comparing the types

File: pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java (hunk 4 above)
Comment (walterddr) [reply pinot-query-planner/src/main/java/org/apache/calcite/rel/hint/PinotHintOptions.java:43]: will do in the next iteration. having boolean method is not that useful IMO b/c there's still 4 combination

File: pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/AggregateNode.java (hunk 5 above)
Comment (walterddr) [reply pinot-query-planner/src/main/java/org/apache/pinot/query/planner/plannode/AggregateNode.java:81]: good idea

**Pull request merged**